jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/LabeledPoint.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j
+
+/** Labeled training data point. */
+private[xgboost4j] case class LabeledPoint(
```

Comment:
Why this class has been made as private while the public constrictor of XMatrix still accepts an Iterator of LabeledPoint? also knowing that it is still valuable to construct a DMatrix and keep it in memory so the process will be able to train many models on the same data set using different combinations of parameters and pick up the best.
on the other hand on using spark interface that accepts dataframe, the conversion to DMatrix will occur for every call and it consumes lot of time for big data.

Reply:
Sorry, I think I have missed that the class is also used in the Java API. 

@CodingCat are you okay with making it public?

Reply:
please go ahead

Reply:
I have created a pull request for this fix, please check https://github.com/dmlc/xgboost/pull/2677